### PR TITLE
if autoloader does not find file, allow next autoloader to try

### PR DESCRIPTION
**Pull request by Flyingmana**
put an 

function a ($c) {
    if($c=='hahahaha'){
        class hahahaha {} // Usually "include 'path/to/file.php';"
    }
}
spl_autoload_register('a');

anywhere in the code (after mageplus autoloader was aktivated)

this autoloader gets never executed, because of the include which is always tried in mageplus autoloader.

This patch makes it possible, to use the typical autoload feature of autoloader chaining.

### CODE REVIEW

**Comment by Flyingmana** [comment]
origin of patch: https://github.com/liip/LiipMagentoBundle/blob/master/magento-autoloader.patch
but very trivial, so using it should be no problem

**Comment by leesbian** [comment]
stream_resolve_include_path requires PHP >= 5.3.2, and only this week I changed the Mage+ requirements to PHP >= 5.3.3 

suggest merging for the v1.2 release, which will give sufficient time for anybody to upgrade

**Comment by LeeSaferite** [comment]
I suggest replacing the stream_resolve_include_path version with a file_exists version for now and then swap to stream_resolve_include_path at some point later.  So we get the fail through benefit now and an increase in speed later.

**Comment by leesbian** [comment]
that sounds like a reasonable compromise

**Comment by Flyingmana** [comment]
and not a problem for me, have changed it to file_exists

File: lib/Varien/Autoload.php [hunk 1]
```diff
@@ -90,6 +90,9 @@ public function autoload($class)
         }
         $classFile.= '.php';
         //echo $classFile;die();
+        if (!file_exists($classFile)) {
+            return false;
+        }
         return include $classFile;
```
Comment (LeeSaferite) [inline lib/Varien/Autoload.php:96]: This line shouldn't be returning the value of the include directly.  It should ignore the include return value, check to verify the class or interface was loaded and return true/false based on that fact.

Including a file that has a 'return' in it causes the call to include to return that value instead of true and this can of course cause problems.

An overall better idea might be to use the ZF autoloader in place of the Varien_Autoload class if possible.

Comment (Flyingmana) [inline lib/Varien/Autoload.php:96]: if you are complain about the "return include $classFile;", this is not really part of the commit, as it is not changed.
But i understand your complain and would suggest, we add a ticket for it, or discuss it on the mailinglist.

Comment (Flyingmana) [inline lib/Varien/Autoload.php:96]: at least, this is nothing what should block this pull request.

Comment (LeeSaferite) [inline lib/Varien/Autoload.php:96]: No, I imagine you are correct.

**Comment by LeeSaferite** [comment]
@leesbian Does this belong in the early stage bug fixes or later stage enhancements?  I'm not sure as it changes the current (buggy) behavior.

**Comment by fabian** [comment]
Also see https://github.com/liip/LiipMagentoBundle/blob/master/magento-autoloader.patch for reference.

**Comment by leesbian** [comment]
@LeeSaferite I see this as an enhancement, rather than a bugfix

I'll merge into the v1.1 branch for testing - I don't believe it will cause any issues, except for where people have manually had to add their own hacks to bootstrap other autoloaders. I think this is an edge case, and those that have already done this will be sufficiently able to resolve any issues (which will likely be reversing their mods, and using the autoloader like they wanted to originally)

**Comment by leesbian** [comment]
merged into develop branch
